docs/src/rules/func-style.md

```diff
@@ -64,6 +64,7 @@ This rule has a string option:
 This rule has an object option for two exceptions:
 
 * `"allowArrowFunctions"`: `true` (default `false`) allows the use of arrow functions. This option applies only when the string option is set to `"declaration"` (arrow functions are always allowed when the string option is set to `"expression"`, regardless of this option)
+* `"allowTypeAnnotation"`: `true` (default `false`) allows the use of function expressions and arrow functions when they have type annotations, regardless of the `allowArrowFunctions` option. This option applies only when the string option is set to `"declaration"`. (TypeScript only)
```

Comment:
```suggestion
* `"allowTypeAnnotation"`: `true` (default `false`) allows the use of function expressions and arrow functions when the variable declaration has type annotation, regardless of the `allowArrowFunctions` option. This option applies only when the string option is set to `"declaration"`. (TypeScript only)
```

I think this would be more accurate.